src/Microsoft.Sbom.Common/Utils/PathUtils.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Linq;
+using Microsoft.Sbom.Common.Config;
+using Microsoft.Sbom.Common.Config.Attributes;
+
+namespace Microsoft.Sbom.Common.Utils;
+
+/// <summary>
+/// Provides utility function to convert an IConfiguration object to use OS-specific path separators.
+/// </summary>
+public static class PathUtils
+{
+    public static void ConvertToOSSpecificPathSeparators(IConfiguration configuration)
+    {
+        var pathProps = configuration.GetType().GetProperties().Where(p => p.GetCustomAttributes(typeof(PathAttribute), true).Any());
```

Comment:
Do we need to walk up the inheritance tree? We can save ourselves some extra reflection if we don't (also chill if you have reason to believe we will need to in the future and we just want to avoid a future bug).

Nitpick: I find it helpful to explicitly provide the parameter name for bare values like bool and int where their purpose isn't immediately obvious. So this would be `p.GetCustomAttributes(typeof(PathAttribute), inherit: true)`.